src/time_integrator.f90

```diff
@@ -39,9 +51,13 @@ function constructor(backend, allocator, nvars)
       constructor%nvars = 3
     end if
 
-    constructor%nolds = 0
+    constructor%istep = 1
+    constructor%order = 1
+    constructor%nolds = constructor%order - 1
```

Comment:
Any objection to rename `constructor` by `init`. This would seem to be more consistent with the rest of the code.